File: src/content/docs/topics/cs/coding-problems/2d-dynamic-programming/518-coin-change-ii.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function change(amount: number, coins: number[]): number {
    // TODO: implement
    return 0;
}

function _runTests(): void {
    assert(change(5, [1, 2, 5]) === 4);
    assert(change(3, [2]) === 0);
    assert(change(0, [1, 2, 5]) === 1);
    assert(change(10, [5]) === 1);
    assert(change(10, [1, 5, 10]) === 4);
    const t0 = performance.now();
    change(1_000_000, [1, 2, 5, 10, 25, 50]);
    console.log(`perf change(1000000, [1,2,5,10,25,50]): ${(performance.now() - t0).toFixed(1)}ms`);
    console.log('all tests pass');
}

_runTests();

File: src/content/docs/topics/cs/coding-problems/2d-dynamic-programming/518-coin-change-ii.go
package main

import "fmt"

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func change(amount int, coins []int) int {
	// TODO: implement
	return 0
}

func runTests() {
	assert(change(5, []int{1, 2, 5}) == 4)
	assert(change(3, []int{2}) == 0)
	assert(change(0, []int{1, 2, 5}) == 1)
	assert(change(10, []int{5}) == 1)
	assert(change(10, []int{1, 5, 10}) == 4)
	fmt.Println("all tests pass")
}

func main() { runTests() }
